lib/bundler.rb

```diff
@@ -352,9 +352,9 @@ def load_gemspec_uncached(file, validate = false)
         spec
       end
     rescue Gem::InvalidSpecificationException => e
-      UI::Shell.new.warn "The gemspec at #{file} is not valid. " \
-        "The validation error was '#{e.message}'"
-      nil
+      UI::Shell.new.warn "The gemspec at #{file} is not valid. Please fix this gemspec.\n" \
+        "The validation error was '#{e.message}'\n"
+      raise Gem::InvalidSpecificationException
```

Comment:
hmmmm. Why do we warn here instead of raising with the full message in the first place?